Lib/test/test_strptime.py

```diff
@@ -406,34 +406,42 @@ def test_offset(self):
-        (*_, offset), _, offset_fraction = _strptime._strptime("-01:30:30", "%z")
-        self.assertEqual(offset, -(one_hour + half_hour + half_minute))
-        self.assertEqual(offset_fraction, 0)
-        (*_, offset), _, offset_fraction = _strptime._strptime("-01:30:30.000001", "%z")
-        self.assertEqual(offset, -(one_hour + half_hour + half_minute))
-        self.assertEqual(offset_fraction, -1)
-        (*_, offset), _, offset_fraction = _strptime._strptime("+01:30:30.001", "%z")
-        self.assertEqual(offset, one_hour + half_hour + half_minute)
-        self.assertEqual(offset_fraction, 1000)
-        (*_, offset), _, offset_fraction = _strptime._strptime("Z", "%z")
-        self.assertEqual(offset, 0)
-        self.assertEqual(offset_fraction, 0)
+        for directive in ("%z", "%:z"):
+            (*_, offset), _, offset_fraction = _strptime._strptime("+01:00",
+                                                                   directive)
+            self.assertEqual(offset, one_hour)
+            self.assertEqual(offset_fraction, 0)
+            (*_, offset), _, offset_fraction = _strptime._strptime("-01:30",
+                                                                   directive)
+            self.assertEqual(offset, -(one_hour + half_hour))
+            self.assertEqual(offset_fraction, 0)
+            (*_, offset), _, offset_fraction = _strptime._strptime("-01:30:30",
+                                                                   directive)
+            self.assertEqual(offset, -(one_hour + half_hour + half_minute))
+            self.assertEqual(offset_fraction, 0)
+            (*_, offset), _, offset_fraction = _strptime._strptime("-01:30:30.000001",
+                                                                   directive)
+            self.assertEqual(offset, -(one_hour + half_hour + half_minute))
+            self.assertEqual(offset_fraction, -1)
+            (*_, offset), _, offset_fraction = _strptime._strptime("+01:30:30.001",
+                                                                   directive)
+            self.assertEqual(offset, one_hour + half_hour + half_minute)
+            self.assertEqual(offset_fraction, 1000)
+            (*_, offset), _, offset_fraction = _strptime._strptime("Z",
+                                                                   directive)
+            self.assertEqual(offset, 0)
+            self.assertEqual(offset_fraction, 0)
 
     def test_bad_offset(self):
-        with self.assertRaises(ValueError):
-            _strptime._strptime("-01:30:30.", "%z")
+        for directive in ("%z", "%:z"):
+            with self.assertRaises(ValueError):
+                _strptime._strptime("-01:30:30.", directive)
+            with self.assertRaises(ValueError):
+                _strptime._strptime("-01:30:30.1234567", directive)
+            with self.assertRaises(ValueError):
+                _strptime._strptime("-01:30:30:123456", directive)
```

Comment:
```suggestion
        error_cases_any_z = [
            "-01:30:30.",         # Decimal point not followed with digits
            "-01:30:30.1234567",  # Too many digits after decimal point
            "-01:30:30:123456",   # Colon as decimal separator
        ]
        for directive in ("%z", "%:z"):
            for timestr in error_cases_any_z:
                with self.subTest(directive, timestr):
                    with self.assertRaises(ValueError):
                        _strptime._strptime(timestr, directive)
```

I suggest parameterizing this a little differently.